server/src/saros/server/session/NegotiationHandler.java

```diff
@@ -146,15 +136,62 @@ public void run() {
   public void handleIncomingProjectNegotiation(
       final AbstractIncomingProjectNegotiation negotiation) {
 
+    Map<String, IProject> projectMapping = new HashMap<>();
+
+    for (ProjectNegotiationData data : negotiation.getProjectNegotiationData()) {
+      String projectName = data.getProjectName();
+      IProject project = workspace.getProject(projectName);
+
+      // TODO: The file path is currently dictated by the name, potentially resulting in CONFLICTS
+      if (!project.exists()) {
+        try {
+          project.adaptTo(ServerProjectImpl.class).create();
+        } catch (IOException e) {
+          projectMapping.clear();
+          negotiation.localCancel(
+              "Error creating project folder", NegotiationTools.CancelOption.NOTIFY_PEER);
+          break;
+        }
+      }
+
+      projectMapping.put(data.getProjectID(), project);
+    }
+
     projectExecutor.execute(
         new Runnable() {
           @Override
           public void run() {
-            negotiation.localCancel(
-                "Server does not accept incoming projects",
-                NegotiationTools.CancelOption.NOTIFY_PEER);
-            negotiation.run(new HashMap<String, IProject>(), new NullProgressMonitor());
+            ProjectNegotiation.Status status;
+            if (ServerConfig.isInteractive()) {
+              status = negotiation.run(projectMapping, new ConsoleProgressIndicator(System.out));
+            } else {
+              status = negotiation.run(projectMapping, new NullProgressMonitor());
+            }
+
+            if (status != ProjectNegotiation.Status.OK)
+              handleErrorStatus(status, negotiation.getErrorMessage(), negotiation.getPeer());
           }
         });
   }
+
+  private void handleErrorStatus(ProjectNegotiation.Status status, String errorMessage, JID peer) {
+    switch (status) {
+      case ERROR:
+        log.error("ERROR running project negotiation: " + errorMessage);
+        break;
+      case REMOTE_ERROR:
+        log.error(
+            "REMOTE_ERROR running project negotiation: "
+                + errorMessage
+                + " at remote: "
+                + peer.toString());
+        break;
+      case CANCEL:
+        log.info("Project negotiation was cancelled locally");
+        break;
+      case REMOTE_CANCEL:
+        log.info("Project negotiation was cancelled by remote: " + peer.toString());
+        break;
```

Comment:
As a suggestion (as this was also noted by Codacy): You could add a default case, either to throw an exception if there is an unhandled case (which would help us notice when an error type got added but this logic was not updated) or to add a catch-all logging statement (could serve the same purpose).